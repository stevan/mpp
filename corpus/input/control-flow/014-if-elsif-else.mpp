if ($x > 10) {
    print("big");
} elsif ($x > 5) {
    print("medium");
} else {
    print("small");
}