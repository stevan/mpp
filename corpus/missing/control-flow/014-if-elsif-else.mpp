} elsif ($x > 5) {
} else {
}