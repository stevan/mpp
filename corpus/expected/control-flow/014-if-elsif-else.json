[
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": ">",
      "left": {
        "type": "Variable",
        "name": "$x"
      },
      "right": {
        "type": "Number",
        "value": "10"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"big\""
          }
        ]
      }
    ],
    "elseIfClauses": [
      {
        "condition": {
          "type": "BinaryOp",
          "operator": ">",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Number",
            "value": "5"
          }
        },
        "block": [
          {
            "type": "Call",
            "name": "print",
            "arguments": [
              {
                "type": "String",
                "value": "\"medium\""
              }
            ]
          }
        ]
      }
    ],
    "elseBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"small\""
          }
        ]
      }
    ]
  }
]